retrieval/targetmanager.go

```diff
@@ -206,18 +299,52 @@ func (ss *scrapeSet) run(ctx context.Context) {
 	}
 
 	wg.Wait()
+
+	ts.sync()
+}
+
+// update handles a target group update from a target provider identified by the name.
+func (ts *targetSet) update(name string, tgroup *config.TargetGroup) error {
+	targets, err := targetsFromGroup(tgroup, ts.config)
+	if err != nil {
+		return err
+	}
+
+	ts.mtx.Lock()
+	defer ts.mtx.Unlock()
+
+	ts.tgroups[name+"/"+tgroup.Source] = targets
+
+	select {
+	case ts.syncCh <- struct{}{}:
+	default:
+	}
+
+	return nil
+}
+
+// scrapePool manages scrapes for sets of targets.
+type scrapePool struct {
+	appender storage.SampleAppender
+
+	ctx     context.Context
```

Comment:
Could be left out in this revision indeed. Will be used with the next PR though.